core/src/test/java/io/questdb/test/std/IntListTest.java

```diff
@@ -151,6 +154,91 @@ public void testSmoke() {
         Assert.assertEquals(0, list.size());
     }
 
+    @Test
+    public void testSortGroups() {
+        Rnd rnd = TestUtils.generateRandom(null);
+        int n = 1 + rnd.nextInt(20);
+        sortGroupsRandom(n, rnd.nextInt(10000), rnd);
+    }
+
+    @Test
+    public void testSortGroupsEqualElements() {
+        Rnd rnd = TestUtils.generateRandom(null);
+        int n = 1 + rnd.nextInt(20);
+        checkSortGroupsEqualElements(n, rnd.nextInt(10000), rnd);
+    }
+
+    @Test
+    public void testSortGroupsNotSupported() {
+        IntList l = new IntList();
+
+        l.add(1);
+        try {
+            l.sortGroups(2);
+            Assert.fail();
+        } catch (IllegalStateException e) {
+            // expected
+        }
+        l.sortGroups(1);
+
+        try {
+            l.sortGroups(-1);
+            Assert.fail();
+        } catch (IllegalStateException e) {
+            // expected
+        }
+    }
+
+    private static void checkSortGroupsEqualElements(int n, int elements, Rnd rnd) {
```

Comment:
The `checkSortGroupsEqualElements` and `sortGroupsRandom` have an identical implementation.

Reply:
true, that's for spotting, fixed in [2622711](https://github.com/questdb/questdb/pull/5232/commits/26227111da66b4f246688cdc705f576e54410b2e)